src/ResourceManagement/Sql/SqlServerAutomaticTuningImpl.cs

```diff
@@ -0,0 +1,159 @@
+namespace Microsoft.Azure.Management.Sql.Fluent
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Core.ResourceActions;
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Core;
+    using Microsoft.Azure.Management.Sql.Fluent.Models;
+    using Microsoft.Azure.Management.Sql.Fluent.SqlServerAutomaticTuning.Update;
+    using Microsoft.Rest;
+    using System.Collections.Generic;
+    using System;
+    using System.Linq;
+
+    /// <summary>
+    /// Implementation for Azure SQL server automatic tuning.
+    /// </summary>
+    ///GENTHASH:Y29tLm1pY3Jvc29mdC5henVyZS5tYW5hZ2VtZW50LnNxbC5pbXBsZW1lbnRhdGlvbi5TcWxTZXJ2ZXJBdXRvbWF0aWNUdW5pbmdJbXBs
+    internal partial class SqlServerAutomaticTuningImpl  :
+        Wrapper<Models.ServerAutomaticTuningInner>,
+        ISqlServerAutomaticTuning,
+        IUpdate
+    {
+        protected string key;
+        private ISqlManager sqlServerManager;
+        private string resourceGroupName;
+        private string sqlServerName;
+
+        ///GENMHASH:BE1AB1D88A6A19EC0BF0334CC3B6BCC1:63EB623539C702D5D92F22479190251D
+        internal SqlServerAutomaticTuningImpl(SqlServerImpl server, ServerAutomaticTuningInner innerObject)
+            : base(innerObject)
+        {
+            this.sqlServerManager = server.Manager;
+            this.resourceGroupName = server.ResourceGroupName;
+            this.sqlServerName = server.Name;
+            this.key = Guid.NewGuid().ToString();
+        }
+
+        ///GENMHASH:670A73E4D654BF4BE6FB621CD7937552:54BF8FD63F8AB586307D57ACE1B3D26B
+        internal SqlServerAutomaticTuningImpl(string resourceGroupName, string sqlServerName, ServerAutomaticTuningInner innerObject, ISqlManager sqlServerManager)
+            : base(innerObject)
+        {
+            this.sqlServerManager = sqlServerManager;
+            this.resourceGroupName = resourceGroupName;
+            this.sqlServerName = sqlServerName;
+            this.key = Guid.NewGuid().ToString();
+        }
+
+
+        ///GENMHASH:17950B8F2EC46B114539E23F9F4D3EA2:EBF69F3A6BD7C5BE989345C27B59C5FE
+        public Microsoft.Azure.Management.Sql.Fluent.Models.AutomaticTuningServerMode DesiredState()
+        {
+            return Inner.DesiredState.GetValueOrDefault();
+        }
+
+        ///GENMHASH:93F5525F475C77754B229151C3005F4B:F84ED1C7DC13C78E0925116541403988
+        public async Task<Microsoft.Azure.Management.Sql.Fluent.ISqlServerAutomaticTuning> ApplyAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var serverAutomaticTuningInner = await this.sqlServerManager.Inner.ServerAutomaticTuning
+                .UpdateAsync(this.resourceGroupName, this.sqlServerName, this.Inner, cancellationToken);
+            this.SetInner(serverAutomaticTuningInner);
+            return serverAutomaticTuningInner != null ? this : null;
```

Comment:
>return serverAutomaticTuningInner != null ? this : null; [](start = 12, length = 56)

This method should return "this".

Reply:
Hmm... I thought I got them all... Fixed now.